### PR TITLE
Add React Native command

### PR DESCRIPTION
**Pull request by JamesLindsay0**
## Goal

Simplify usage when uploading react native source maps.

## Design

The command accepts some IOS or Android specific data such as the variant using `android-variant`.

Options that can't be set through the top level command are:
- `--application-id`
- `--android-ndk-root`
- `--build-uuid`
- `--no-build-uuid`
- `--dex-files`
- `--ignore-empty-dsym`
- `--ignore-missing-dwarf`

## Changeset

New command added

## Testing

A maze runner test.

### CODE REVIEW

File: pkg/upload/react-native.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/options"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+)
+
+func ProcessReactNative(globalOptions options.CLI, endpoint string, logger log.Logger) error {
+	reactNativeOptions := globalOptions.Upload.ReactNative
+
+	// Some commands must be run from either the android/ or ios/ subdirectory.
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:14]: I believe the two source map commands can be run from inside their respective platform directories, so perhaps use these paths in each command for consistency?

File: pkg/upload/react-native.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	androidPath := []string{}
+	iosPath := []string{}
+	for _, basePath := range reactNativeOptions.Path {
+		androidPath = append(androidPath, filepath.Join(basePath, "android"))
+		iosPath = append(iosPath, filepath.Join(basePath, "ios"))
+	}
+
+	logger.Info("Uploading React Native Android")
+	globalOptions.Upload.ReactNativeAndroid = options.ReactNativeAndroid{Path: reactNativeOptions.Path, ProjectRoot: reactNativeOptions.ProjectRoot, ReactNative: reactNativeOptions.Shared, Android: reactNativeOptions.AndroidSpecific}
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:23]: Formatting

File: pkg/upload/react-native.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	if err := ProcessReactNativeAndroid(globalOptions, endpoint, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Uploading React Native iOS")
+	globalOptions.Upload.ReactNativeIos = options.ReactNativeIos{Path: reactNativeOptions.Path, ProjectRoot: reactNativeOptions.ProjectRoot, ReactNative: reactNativeOptions.Shared, Ios: reactNativeOptions.IosSpecific}
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:29]: Formatting

File: pkg/upload/react-native.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		iosPath = append(iosPath, filepath.Join(basePath, "ios"))
+	}
+
+	logger.Info("Uploading React Native Android")
+	globalOptions.Upload.ReactNativeAndroid = options.ReactNativeAndroid{Path: reactNativeOptions.Path, ProjectRoot: reactNativeOptions.ProjectRoot, ReactNative: reactNativeOptions.Shared, Android: reactNativeOptions.AndroidSpecific}
+	if err := ProcessReactNativeAndroid(globalOptions, endpoint, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Uploading React Native iOS")
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:28]:
```suggestion
	logger.Info("Uploading JavaScript source maps for iOS")
```

File: pkg/upload/react-native.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+	// Some commands must be run from either the android/ or ios/ subdirectory.
+	androidPath := []string{}
+	iosPath := []string{}
+	for _, basePath := range reactNativeOptions.Path {
+		androidPath = append(androidPath, filepath.Join(basePath, "android"))
+		iosPath = append(iosPath, filepath.Join(basePath, "ios"))
+	}
+
+	logger.Info("Uploading React Native Android")
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:22]:
```suggestion
	logger.Info("Uploading JavaScript source maps for Android")
```

File: pkg/upload/react-native.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		return err
+	}
+
+	logger.Info("Uploading React Native iOS")
+	globalOptions.Upload.ReactNativeIos = options.ReactNativeIos{Path: reactNativeOptions.Path, ProjectRoot: reactNativeOptions.ProjectRoot, ReactNative: reactNativeOptions.Shared, Ios: reactNativeOptions.IosSpecific}
+	if err := ProcessReactNativeIos(globalOptions, endpoint, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Uploading Android NDK")
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:34]:
```suggestion
	logger.Info("Uploading Android NDK symbols")
```

File: pkg/upload/react-native.go [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+		ProjectRoot: reactNativeOptions.ProjectRoot,
+		AppManifest: reactNativeOptions.AndroidSpecific.AppManifest,
+		Variant:     reactNativeOptions.AndroidSpecific.Variant,
+		VersionCode: reactNativeOptions.AndroidSpecific.VersionCode,
+	}
+	if err := ProcessAndroidNDK(globalOptions, endpoint, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Uploading Android Proguard")
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:48]:
```suggestion
	logger.Info("Uploading Android Proguard mappings")
```

File: pkg/upload/react-native.go [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+		VersionName: reactNativeOptions.Shared.VersionName,
+		AppManifest: reactNativeOptions.AndroidSpecific.AppManifest,
+		Variant:     reactNativeOptions.AndroidSpecific.Variant,
+		VersionCode: reactNativeOptions.AndroidSpecific.VersionCode,
+	}
+	if err := ProcessAndroidProguard(globalOptions, endpoint, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Uploading DSYM")
```
Comment (tomlongridge) [inline pkg/upload/react-native.go:61]:
```suggestion
	logger.Info("Uploading iOS dSYMs")
```

File: features/react-native/rn0_72.feature
Comment (tomlongridge) [inline features/react-native/rn0_72.feature:1]: Can we get a better test off the different file types from this?

We might need some input from @joshedney on how we can get the appropriate symbol files - i.e. dSYMs and Android mappings - enabled in the project so we can upload them?

Comment (JamesLindsay0) [reply features/react-native/rn0_72.feature:1]: I've added those to the test.

**Review by joshedney — APPROVED** [review]
LGTM!

**Pull request merged**